test/com/amazon/ion/impl/lite/BaseIonSequenceLiteSublistTestCase.java

```diff
@@ -60,6 +66,40 @@ public void sublistIsEmpty() {
         assertFalse(sequence.subList(0, 1).isEmpty());
     }
 
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void subListFromIndexLessThanZero() {
+        newSequence().subList(-1, 1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void subListToIndexLessThanFromIndex() {
+        newSequence().subList(2, 1);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void subListToIndexExceedsSize() {
+        IonSequence seq = newSequence();
+        // toIndex is exclusive, hence the + 1
+        seq.subList(0, seq.size() + 1);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void subListOfSubListFromIndexLessThanZero() {
+        newSubList().subList(-1, 1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void subListOfSubListToIndexLessThanFromIndex() {
+        newSubList().subList(2, 1);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void subListOfSubListToIndexExceedsSize() {
+        List<IonValue> seq = newSubList();
+        // toIndex is exclusive, hence the + 1
+        seq.subList(0, seq.size() + 1);
+    }
+
```

Comment:
Do we have similar tests around non-empty sequences?

Reply:
These are non-empty sequences.  Would you like me to add some for some emtpy sequences too?

Reply:
Yeah, my misunderstanding about the test data.

I think the empty case is kind of interesting, but I will leave it to you if you think it is worth it.  I mean for bonus points we'd fuzz test this, but I think that to be overkill.